pupil-spa/src/app/services/user/user.service.ts

```diff
@@ -14,25 +14,19 @@ export class UserService {
   private loggedIn = false;
   data: any = {};
 
-  constructor(private http: Http, private storageService: StorageService) {
+  constructor(private http: HttpClient, private storageService: StorageService) {
     this.loggedIn = !!this.storageService.getItem(accessTokenKey);
   }
 
   login(schoolPin, pupilPin): Promise<any> {
     return new Promise(async (resolve, reject) => {
-      const headers = new Headers();
-      headers.append('Content-Type', 'application/json');
-      const requestArgs = new RequestOptions({headers: headers});
+      const headers = {
+        headers: new HttpHeaders( { 'Content-Type': 'application/json' })
+      };
 
-      await this.http.post(`${APP_CONFIG.authURL}`,
-        { schoolPin, pupilPin },
-        requestArgs)
+      await this.http.post(`${APP_CONFIG.authURL}`, { schoolPin, pupilPin }, headers)
         .toPromise()
-        .then((response) => {
-          if (response.status !== 200) {
-            return reject(new Error('Login Error:' + response.status + ':' + response.statusText));
-          }
-          const data = response.json();
+        .then(data => {
```

Comment:
how does the app handle rejection with this change?
Was that block unnecessary in the first place?

Reply:
StatusCode is no longer provided when using httpClient - so it a 401 / 500 will be rejected.

Reply:
I see, will the rejection be handled then fine?
What will the user view in the frontend when it occurs?